pkgs/applications/misc/dockbarx/default.nix

```diff
@@ -1,40 +1,77 @@
+  version = "${ver}-${rev}";
+  ver = "1.0-beta";
+  rev = "d98020e";
 
   src = fetchFromGitHub {
-    owner = "M7S";
+    owner = "xuzhen";
     repo = "dockbarx";
-    rev = ver;
-    sha256 = "1h1g2vag5vnx87sa1f0qi8rq7wlr2ymvkrdr08kk7cma4wk0x6hg";
+    rev = rev;
+    sha256 = "0xwqxh5mr2bi0sk54b848705awp0lfpd91am551811j2bdkbs04m";
   };
 
+  nativeBuildInputs = [
+    glib
+    python3Packages.polib
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    gobject-introspection
+    gtk3
+    libwnck3
+    keybinder3
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
+    dbus-python
+    pillow
+    pygobject3
+    pyxdg
+    xlib
+  ];
+
+  # no tests
+  doCheck = false;
+
+  dontWrapGApps = true;
+
   postPatch = ''
     substituteInPlace setup.py                                --replace /usr/                   ""
     substituteInPlace setup.py                                --replace '"/", "usr", "share",'  '"share",'
+    substituteInPlace dbx_preference                          --replace /usr/share/             $out/share/
     substituteInPlace dockbarx/applets.py                     --replace /usr/share/             $out/share/
     substituteInPlace dockbarx/dockbar.py                     --replace /usr/share/             $out/share/
     substituteInPlace dockbarx/iconfactory.py                 --replace /usr/share/             $out/share/
     substituteInPlace dockbarx/theme.py                       --replace /usr/share/             $out/share/
-    substituteInPlace dockx_applets/battery_status.py         --replace /usr/share/             $out/share/
-    substituteInPlace dockx_applets/namebar.py                --replace /usr/share/             $out/share/
-    substituteInPlace dockx_applets/namebar_window_buttons.py --replace /usr/share/             $out/share/
-    substituteInPlace dockx_applets/volume-control.py         --replace /usr/share/             $out/share/
+    substituteInPlace mate_panel_applet/dockbarx_mate_applet  --replace /usr/share/             $out/share/
   '';
 
-  propagatedBuildInputs = (with python2Packages; [ pygtk pyxdg dbus-python pillow xlib ])
-    ++ (with gnome2; [ gnome_python gnome_python_desktop ])
-    ++ [ keybinder ];
+  postInstall = ''
+    ${glib.dev}/bin/glib-compile-schemas "$out/share/glib-2.0/schemas"
+  '';
 
-  # no tests
-  doCheck = false;
+  # Arguments to be passed to `makeWrapper`, only used by buildPython*
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
 
   meta = with lib; {
-    homepage = "https://launchpad.net/dockbar/";
-    description = "Lightweight taskbar / panel replacement for Linux which works as a stand-alone dock";
-    license = licenses.gpl3;
+    homepage = "https://github.com/xuzhen/xfce4-dockbarx-plugin";
+    description = "Lightweight taskbar/panel replacement which works as a stand-alone dock";
+    license = licenses.gpl3Only;
     platforms = platforms.linux;
     maintainers = [ maintainers.volth ];
```

Comment:
I think we should replace volth here as he is no longer active.

Reply:
Done.